packages/video_player/video_player/lib/video_player.dart

```diff
@@ -460,7 +460,9 @@ class VideoPlayerController extends ValueNotifier<VideoPlayerValue> {
           // position=value.duration. Instead of setting the values directly,
           // we use pause() and seekTo() to ensure the platform stops playing
           // and seeks to the last frame of the video.
-          pause().then((void pauseResult) => seekTo(value.duration));
+          if (value.isPlaying) {
+            pause().then((void pauseResult) => seekTo(value.duration));
+          }
```

Comment:
This changed logic would definitely not be guaranteed to actually do what this code is supposed to be doing; if the native side reported reaching the end as:
- playback stopped
- video completed

then the seek code would never run.

Reply:
@stuartmorgan have updated the logic to account for this